docs/guides/applications-with-stackables.md

```diff
@@ -84,56 +80,121 @@ foo.configManagerConfig = {
+
+```js
+class Generator extends ServiceGenerator {
+  getDefaultConfig () {
+    const defaultBaseConfig = super.getDefaultConfig()
+    const defaultConfig = {
+      greeting: 'Hello world!'
+    }
+    return Object.assign({}, defaultBaseConfig, defaultConfig)
+  }
+
+  async _getConfigFileContents () {
+    const baseConfig = await super._getConfigFileContents()
+    const config = {
+      $schema: './stackable.schema.json',
+      greeting: {
+        text: this.config.greeting ?? 'Hello world!'
+      }
+    }
+    return Object.assign({}, baseConfig, config)
+  }
+
+  async _afterPrepare () {
+    this.addFile({
+      path: '',
+      file: 'stackable.schema.json',
+      contents: JSON.stringify(schema, null, 2)
+    })
+  }
+}
+```
+
+This generator extends `ServiceGenerator` which means it will generate an application based on Platformatic Service. If you want to create a Stackable based on Platformatic DB or Platformatic Composer you can extend `DBGenerator` or `ComposerGenerator` respectively.
+
+In addition this generator adds a custom `greeting` property to the Stackable config and generates a `stackable.schema.json` file with the JSON schema of the Stackable config.
+
+### Stackable schema
 
-module.exports = foo
+`schema` is the JSON schema of the Stackable config. It's used to validate the Stackable config. You can find an example of a schema in `lib/schema.js`.
+
+```js
+const stackableSchema = {
+  ...schema.schema,
+  $id: 'stackable',
+  title: 'Stackable Config',
+  properties: {
+    ...schema.schema.properties,
+    greeting: {
+      type: 'object',
+      properties: {
+        text: {
+          type: 'string'
+        }
+      },
+      required: ['text'],
+      additionalProperties: false
+    }
+  },
+}
 ```
 
-Note that the `$id` property of the schema identifies the module in our system,
-allowing us to retrieve the schema correctly.
-It is recommended, but not required, that the JSON schema is actually
-published in this location. Doing so allows tooling such as the VSCode
-language server to provide autocompletion.
+This schema extends the Platformatic Service schema and adds a custom `greeting` property. You can also use the stackable schema to generate a types file for the Stackable config. To do so, you can use `npm run build:config` cli command.
 
-In this example, the `schema` adds a custom top-level `foo` property
-that users can use to configure this specific module.
+### Creating a Stackable application
 
-ESM is also supported.
+If you chose to use TypeScript, you will need to run `npm run build` first to compile the Stackable.
 
-## Consuming a custom application
+To create an application based on a Stackable you can run the `npm run create` cli command. This command uses the `cli/create.js` script to generate the application. By default application will be generated in the `./app` folder. As a next step you need to install the dependencies.
```

Comment:
`npm run create` sounds a bit odd. I think you want to frame it as

1. publish the module
2. do a global install
3. run the `stackable` command

(the publish/install could be replace with linking

Reply:
When I'm developing a stackable I want to test from time to time. The esiest way to do this it to generate a application. If I understand you correctly I would need to publish a stackable every time I make a change to be able to test it. Did I understand you correctly?

Reply:
I wouldn't be surprized if 90% of people that would try it don't have an npm account to publish.

Reply:
`npm link` is a thing.

Also running `node cli/create.js --dir path/to/myapp` is way nicer.
`cli/create.js` should also be set in the bin scripts.

Reply:
> `npm link` is a thing.

I mean I don't understand where I can link it. I have a generated stackable project and globally installed platformatic. What it the next step?

Reply:
`npm link --global` would make the bins available globally

Reply:
> Also running `node cli/create.js --dir path/to/myapp` is way nicer.

npm run create executes commmand exactly like this
https://github.com/platformatic/platformatic/pull/2016/files#diff-90f9d2ea47f8756b57ede9621e150c414c1a5963ac1b5c33825fb03942262c72R242

Reply:
I would not recommend running a generator inside the same folder I'm currently working on. It just creates a local mess, which could possibly ruin what I'm currently working on. It's better to run it in an external folder.